src/pluginlaunch.c

```diff
@@ -398,14 +398,20 @@ plugin_timeout (nvti_t *nvti)
 static int
 get_available_memory ()
 {
-  char buf[8192], *hit;
-  FILE *fd;
+  char buf[8192] = NULL;
+  char *hit = NULL;
+  FILE *fd = NULL;
   size_t len;
 
   fd = fopen ("/proc/meminfo", "r");
+  if (fd == NULL)
+    {
+      g_warning ("Couldn't open /proc/meminfo");
+      return 0;
+    }
   len = fread (buf, 1, sizeof (buf) - 1, fd);
   fclose (fd);
-  if (len == 0)
+  if (len < (sizeof (buf) - 1) || buf == NULL)
```

Comment:
If the content of the file has fewer chars than the buffer size, should not be an error.
Actually, I have just checked with `wc -c /proc/meminfo` and I got 1503 chars.

Reply:
Hi, @jjnicola 
Thank you for review.

Oh I see, i ii revert it.

haha thanks 👍 
> Actually, I have just checked with wc -c /proc/meminfo and I got 1503 chars.